.github/workflows/wheels-manylinux-publish.yml

```diff
@@ -0,0 +1,68 @@
+name: Build RAPIDS manylinux wheels
+
+on:
+  workflow_call:
+    inputs:
+      # repo and branch
+      repo:
+        type: string
+      branch:
+        type: string
+      date:
+        type: string
+      sha:
+        type: string
+      build_type:
+        required: true
+        type: string
+
+      # general settings
+      package-name:
+        required: true
+        type: string
+
+      # twine settings
+      build-tag:
+        required: false
+        type: string
+        default: ''
+
+jobs:
+  wheel-publish:
+    name: wheels publish all wheels
+    runs-on: ubuntu-latest
+    container:
+      # ctk version of the cibw container is irrelevant in the publish step
+      # it's simply a launcher for twine
+      image: "rapidsai/cibuildwheel:cuda-runtime-11.8.0-ubuntu20.04"
+      env:
+        RAPIDS_PY_WHEEL_BUILD_TAG: ${{ inputs.build-tag }}
+        RAPIDS_BUILD_TYPE: ${{ inputs.build_type }}
+        AWS_ACCESS_KEY_ID: ${{ secrets.RAPIDSAI_GHA_AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.RAPIDSAI_GHA_AWS_SECRET_ACCESS_KEY }}
+        TWINE_USERNAME: cibuildwheel
+        TWINE_PASSWORD: ${{ secrets.RAPIDSAI_PYPI_CI_PASSWORD }}
+        TWINE_REPOSITORY_URL: "https://pypi.k8s.rapids.ai/simple/"
+    steps:
+    - name: Standardize repository information
+      run: |
+        echo "RAPIDS_REPOSITORY=${{ inputs.repo || github.repository }}" | tee --append "${GITHUB_ENV}" "${GITHUB_OUTPUT}"
+        sha="${{ inputs.sha }}"
+        if [[ "${sha}" == "" ]]; then
+          sha=$(git rev-parse HEAD)
+        fi
+        echo "RAPIDS_SHA=${sha}" | tee --append "${GITHUB_ENV}" "${GITHUB_OUTPUT}"
+        echo "RAPIDS_REF_NAME=${{ inputs.branch || github.ref_name }}" | tee --append "${GITHUB_ENV}" "${GITHUB_OUTPUT}"
+        echo "RAPIDS_NIGHTLY_DATE=${{ inputs.date }}" | tee --append "${GITHUB_ENV}" "${GITHUB_OUTPUT}"
+
+    - name: Download wheels from downloads.rapids.ai and publish to internal PyPI
+      run: |
```

Comment:
Should we implement [a tool like `rapids-upload-to-anaconda`](https://github.com/rapidsai/gha-tools/blob/main/tools/rapids-upload-to-anaconda)? Maintaining the matrix in the form of a bunch of repeated lines like this isn't great and would be better suited to a script where we have true loop constructs available to handle that.

Reply:
My thoughts on this are mixed;

The matrix for building (pyver, arch) is decided in this workflow. The matrix for publishing corresponds to the matrix for building.

E.g. to change py 3.9 to 3.10 would be a sed command of 's/3.9/3.10/g' in the wheel workflow files in this repo, without involving gha-tools.

Reply:
I just used GHA's built-in matrix again. This way the same matrix reflects across build, test, publish.